UnityProject/Assets/Scripts/Shuttles/EscapeShuttleConsole.cs

```diff
@@ -72,17 +83,70 @@ private void FinishHack(HandApply interaction)
 
 			beenEmagged = true;
 
-			if (GameManager.Instance.ShuttleSent) return;
+			DepartShuttle();
+		}
+
+		private void RegisterEarlyShuttleLaunch(IDCard card, PlayerScript performer)
+		{
+			if (GameManager.Instance.ShuttleSent)
+			{
+				Chat.AddExamineMsg(performer.gameObject, "The shuttle is already moving!");
+				return;
+			}
+			if (registeredIDs.Contains(card))
+			{
+				Chat.AddExamineMsg(performer.gameObject, "You've already done this!");
+				return;
+			}
+			registeredIDs.Add(card);
+
+			if (GameManager.Instance.CentComm.CurrentAlertLevel is CentComm.AlertLevel.Red or CentComm.AlertLevel.Delta && registeredIDs.Count >= 2)
+			{
+				DepartShuttle();
+				return;
+			}
+
+			if (registeredIDs.Count >= 4)
+			{
+				DepartShuttle();
+				return;
+			}
+
+			AnnounceRemainingSwipesRequired();
+		}
+
+		private void AnnounceRemainingSwipesRequired()
+		{
+			var swpiesRequired = GameManager.Instance.CentComm.CurrentAlertLevel is CentComm.AlertLevel.Red or CentComm.AlertLevel.Delta ? 2 : 4;
```

Comment:
The same logic is used above, refactor it into a method.

Reply:
?

Reply:
The logic that calculates whether 2 or 4 ID cards are required is in two places. It should be defined in one spot.